internal/connect/zypper.go

```diff
@@ -244,3 +254,99 @@ func DistUpgrade(version string, quiet, verbose, nonInteractive bool, extraArgs
 	_, err := zypperRun(args, []int{zypperOK})
 	return err
 }
+
+func createZypperTarball(tarballPath, root string, paths []string) error {
+	// tar reports an error if a file does not exist.
+	// So we have to check this before.
+	var existingPaths []string
+	for _, p := range paths {
+		if !fileExists(p) {
+			continue
+		}
+		// remove leading "/" from paths to allow using them from different root
+		existingPaths = append(existingPaths, strings.TrimLeft(p, "/"))
+	}
+
+	// ensure directory exists
+	if err := os.MkdirAll(path.Dir(tarballPath), os.ModeDir); err != nil {
+		return err
+	}
+
+	command := []string{"tar", "cz", "-C", root, "-f", tarballPath}
+	command = append(command, existingPaths...)
+	_, err := execute(command, []int{0})
+
+	// tarball can contain sensitive data, so prevent read to non-root
+	// do it for sure even if tar failed as it can contain partial content
+	if fileExists(tarballPath) {
+		os.Chmod(tarballPath, 0600)
+	}
+
+	if err != nil {
+		return fmt.Errorf("failed to create backup")
+	}
+	return nil
+}
+
+func createZypperRestoreScript(scriptPath, tarballPath string, paths []string) error {
+	var data struct {
+		Paths   []string
+		Tarball string
+	}
+	// remove leading "/" from paths to allow using them from different root
+	for _, p := range paths {
+		data.Paths = append(data.Paths, strings.TrimLeft(p, "/"))
+	}
+	data.Tarball = strings.TrimLeft(tarballPath, "/")
```

Comment:
so the restore script uses a relative path. But the backup always puts it in `/var/adm/backup/system-upgrade/repos.tar.gz`

Reply:
Yeah, there are some bugs around `--root` in the original. I've already fixed one here: https://github.com/SUSE/zypper-migration/pull/14
I'll re-check it again.

Reply:
I updated the code to store both script and tarball under given `--root`. Tested and seem to be working fine.
I added a note in DIFFERENCES as this seems to be broken in the original code.

Reply:
ok thanks